rts/Sim/Misc/Team.cpp

```diff
@@ -349,8 +349,15 @@ void CTeam::SlowUpdate()
 			if (team->isDead)
 				continue;
 
-			const float edif = std::max(0.0f, (team->resStorage.energy * 0.99f) - team->res.energy) * de;
```

Comment:
`const float edif = Clamp(((team->resStorage.energy * 0.99f) - team->res.energy) * de, 0.0f, res.energy);`
`const float mdif = Clamp(((team->resStorage.metal * 0.99f) - team->res.metal) * dm, 0.0f, res.metal);`

Don't keep the old lines.

Reply:
ok, will change.